uniffi/src/bindings/kotlin/templates/RustBufferHelpers.kt

```diff
@@ -164,8 +164,8 @@ fun Double.lowerInto(buf: ByteBuffer) {
     buf.putDouble(this)
 }
 
-fun String.lower(): String {
-    return this
+fun String.lower(): Pointer {
+    return _UniFFILib.INSTANCE.{{ ci.ffi_string_alloc_from().name() }}(this)
```

Comment:
I like it!
One clarification (in case I'm missing something 😆 ):

Here's what the process looks like in my head:
- Kotlin wants to call todo.addItem("Make a review comment!")
- Kotlin then will call `String.lower()`, which in turn calls `string_alloc` to create a Rust allocation for the String
- Kotlin will pass that Rust owned string to the `addItem` function across the FFI
- The pointer is back in rust land, and Rust re-takes ownership of that string and appropriately drops it when it goes out of scope
- The function is done, we are happy and all memory used is freed.

Sound about right? Asking because I got confused there for a second thinking that strings created here would need an explicit `free` once the function is done, but that is covered by Rust re-taking the ownership right?

_Wonders if there is a scenario where things can go wrong in between the allocation and the function call causing a leak_

Reply:
> Sound about right? 

Yes, exactly. This is similar to how lowering into a buffer works, where e.g. `Record.lower()` allocates and fills a buffer, passes the buffer to the rust code, and relies on the rust code to free it.

Which...now that you point it out, we don't actually do, because [`ByteBuffer` doesn't implement `Drop`](https://docs.rs/ffi-support/0.4.0/ffi_support/struct.ByteBuffer.html#caveats) 🤦.

I'll file a separate bug to dig into that.

Reply:
I filed https://github.com/mozilla/uniffi-rs/issues/194 about freeing bytebuffers, but I think we may actually be OK due to the way we access the underlying data.

I also filed https://github.com/mozilla/uniffi-rs/issues/195 about the possibility of leaking memory if we get interrupted part-way through lowering arguments to pass over the FFI.